ee/desktop/menu/dark_mode_darwin.go

```diff
@@ -0,0 +1,48 @@
+//go:build darwin
+// +build darwin
+
+package menu
+
+/*
+#cgo CFLAGS: -x objective-c
+#cgo LDFLAGS: -framework Foundation
+
+#include <AppKit/AppKit.h>
+
+bool isDarkMode();
+void watchTheme();
+*/
+import (
+	"C"
+)
+import (
+	"sync"
+)
+
+var (
+	themeChangeListeners []func()
+	once                 sync.Once
+)
+
+func isDarkMode() bool {
+	if C.isDarkMode() {
+		return true
+	}
+	return false
+}
+
+// Registers a listener to be notified when OS theme (dark/light) changes
+func RegisterThemeChangeListener(f func()) {
+	themeChangeListeners = append(themeChangeListeners, f)
+
+	// Ensure we register ourselves as an observer only once
+	watchTheme := func() { C.watchTheme() }
+	once.Do(watchTheme)
+}
+
+//export themeChanged
```

Comment:
What does `export` mean here?

Reply:
Yeah not sure if this is a Go idiom, but I meant to call out that this function is marked as `extern`, so it's callable from C code, and in a sense exported to C. I can remove if it's not clear, since Go has it's own concept of exporting symbols.